GUI/MainWindow.xaml.cs

```diff
@@ -1255,6 +1258,17 @@ private bool SpectraFileExists(ObservableCollection<RawDataForDataGrid> rDOC, Ra
             return false;
         }
 
+        private void CancelButton_Click(object sender, RoutedEventArgs e)
+        {
+            string grammar = StaticTasksObservableCollection.Count <= 1 ? "this task" : "these tasks";
+            if (MessageBox.Show("Are you sure you want to cancel " + grammar + "?", "Cancel Tasks", MessageBoxButton.OKCancel) == MessageBoxResult.OK)
+            {
+                GlobalVariables.StopLoops = true;
+                CancelButton.IsEnabled = false;
+                notificationsTextBox.AppendText("Canceling...\n");
```

Comment:
I like this, but it didn't go away after it finished when I canceled during protein loading. Maybe another message saying "Canceled!" or just remove the "Canceling..." once it's finished?